src/base/scene.cu

```diff
@@ -0,0 +1,309 @@
+#include <stack>
+
+#include "scene.h"
+#include "scene_object.h"
+
+#define TINYGLTF_IMPLEMENTATION
+#define STB_IMAGE_IMPLEMENTATION
+#include "tiny_gltf.h"
+
+__host__ gm::Scene::Scene(const std::string &filepath) {
+  tinygltf::Model model;
+  tinygltf::TinyGLTF loader;
+  std::string err;
+  std::string warn;
+
+  std::ifstream fileStream;
+
+  fileStream.open(filepath);
+  if (fileStream.fail()) {
+    std::cout << "Unable to open file: " << filepath << std::endl;
+  }
+
+  static const uint8_t HEADER_BYTES = 4;
+
+  char magicNumber[HEADER_BYTES];
+  fileStream.read(magicNumber, HEADER_BYTES);
+  fileStream.close();
+
+  bool ret = false;
+  // Check magic number 0x46546c67 == "glTF"
+  if (*reinterpret_cast<int *>(magicNumber) == 0x46546c67) {
+    ret = loader.LoadBinaryFromFile(&model, &err, &warn, filepath);
+  } else if (filepath.substr(filepath.length() - 5, filepath.length()) ==
+             ".gltf") {
+    ret = loader.LoadASCIIFromFile(&model, &err, &warn, filepath);
+  } else {
+    // Error here because we got an unknown file extension.
+    // TODO error handling
+  }
+
+  if (!warn.empty()) {
+    printf("Warn: %s\n", warn.c_str());
```

Comment:
Small nitpick, but we are mixing C++ and C style print statements. I'd stick to the C++ style for now just to be consistent